### ISSUE TITLE
Low GPU usage?

### ISSUE THREAD

**Issue by tly627** [DeepLabCut/DeepLabCut#1485]
Hello, I am new to DeepLabCut and I have a question about GPU.  I have followed the instructions to install all the required software for GPU support and have made sure that their versions are compatible with each other according to these tables: https://www.tensorflow.org/install/source_windows
https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html.

It seems like DeepLabCut can find my GPU device (see attached) when I start the training but when I looked at task manager, Python is always using only 2.3-2.4% of the GPU and around 25-30% of CPU. I am wondering if this is normal. Also, it seems like my computer is taking ~17 secs to do 100 iterations (~9.4 hrs for 200,000 iterations). Is this speed slower than expected for RTX 3070?

![tensorflow](https://user-images.githubusercontent.com/63482670/131052712-2a57662e-219f-4883-8bcb-8fb40d930a3b.PNG)
![GPU usage](https://user-images.githubusercontent.com/63482670/131052718-caa8591e-d5cd-4dde-adfb-37e504d7e461.PNG)

### Operating System

operating system: Windows 10
NVIDIA driver: 461.40
CUDA: 11.2.2
cudnn: 8.1.0.77
Tensorflow: 2.5.0

### DeepLabCut version

dlc version 2.2.0.1

### DeepLabCut mode

single animal

### Device type

gpu: NVIDIA GeForce RTX 3070

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by KonradDanielewski** [comment]
Depends on the chosen architecture, batchsize, image size. Seems reasonable. From software point of view, tensorflow device is created, memory gets allocated and it does run on the GPU so everyhing looks fine.

**Comment by tly627** [comment]
Good to know that it's actually working. Thanks for the confirmation!

**Comment by KonradDanielewski** [comment]
Short explanation for the future viewers of this issue. GPU usage is shown based on some internal calculations that might differ between different die architectures, drivers, workloads etc. A well optimized workload might show low GPU usage as in, at given time the resources (CUDA cores for instance) are utilized in 2-3% but when you look at the memory usage, with DLC (more specifically it's tensorflow that allocates it) the memory will be almost entirely allocated (doesn't have to be used entirely but it's reserved for this workload anyway). So in reality, though a lot of memory is being used, calculations performed on data located in this memory are very efficient (and might be bottlenecked by the memory bus - how much data can be transferred from memory to the GPU to be processed) and don't need to use more than the 2-3% of your cores. 

To sum up, what you should look at if you're not sure if your training/inference is being done on the GPU look at the GPU memory usage and if the Tensorflow device was created using GPU_device (printed when running e.g. training)

**Comment by backyardbiomech** [comment]
I think this would be another good page for the cookbook (“How to tell if deeplabcut is fully using your GPU”) since it has come up repeatedly.

**Comment by KonradDanielewski** [comment]
> I think this would be another good page for the cookbook (“How to tell if deeplabcut is fully using your GPU”) since it has come up repeatedly.

Since I'll be doing a DLC workshop for some people in 2-3 weeks I might try to write all of this down for the cookbook as part of workshop preparations. If you (or anybody else) think there is some other thing that could be useful let me know, I might have some free time to do this after I finish current experiment.

**Comment by MMathisLab** [comment]
@backyardbiomech any interest in making your excellent suggestion of a how to tell your GPU is used recipe :) ?

**Comment by backyardbiomech** [comment]
Happy to, but I’m going on vacation for a week today, so it will have to wait.
